pkg/util/pool/bucketed_pool.go

```diff
@@ -100,3 +100,11 @@ func (p *BucketedPool[T, E]) Put(s T) {
 func IsPowerOfTwo(n int) bool {
 	return (n & (n - 1)) == 0
 }
+
+func NextPowerTwo(n int) int {
```

Comment:
If this is needed elsewhere, we should probably move it to a different package. `pkg/util/math` perhaps?